### DISCUSSION TITLE
Latency: Feeling laggy when using on windows

### DISCUSSION THREAD

**Discussion by Parsifa1** [ibhagwan/fzf-lua#1353] [Q&A] [answered]
![recording](https://github.com/user-attachments/assets/d9c40596-2e5c-42b8-930c-61a589532f38)

It seems that when used on windows, there is some delay in opening and closing the window(The window appears first, then the text)
I have observed that the adaptation of fzf-lua to windows seems to have not been completely completed, so is this problem expected?
I don’t mean to criticize anything, I’m just curious about the specific progress. There doesn’t seem to be an issue where I can see the progress of the entire windows support.

**Comment by ibhagwan** [comment, marked as answer]
> It seems that when used on windows, there is some delay in opening and closing the window(The window appears first, then the text)
I have observed that the adaptation of fzf-lua to windows seems to have not been completely completed, so is this problem expected?
> I don’t mean to criticize anything, I’m just curious about the specific progress. There doesn’t seem to be an issue where I can see the progress of the entire windows support.

Windows was completed, unforutnely the lag in opening is expected, you can try `fzf.vim` it’s exactly the same, the issue stems from having to open fzf inside a terminal using `cmd.exe`, just try `:new | term` it would be roughly the same speed, not much I can do about it.

**EDIT:** It also might be related to windows defender, some users are reporting tremendous boost after excluding the folder/executables (neovim, fzf, rg, etc)

**Reply by Parsifa1** [reply]
Thank you very much for creating this beautiful plugin, I mostly use it under WSL, so it's still fast :)

**Reply by ibhagwan** [reply]
Ty for the kind words <3

> I mostly use it under WSL, so it's still fast :)

That was my experience too when developing Windows support :)

**Comment by Parsifa1** [comment]
You might think I'm being nitpicky, but I'm wondering if this is possible:
During the initialization process of fzf-lua, there will be a momentary flashing of the cursor. Is there a way for us to set `TermCursor` to `NONE` when initializing the terminal buffer, and set `TermCursor` back before fzf starts (at that point in time). my eyes always notice the cursor flickering strangely (Appears in the upper left corner first, then where it should be)
 :(

![recording](https://github.com/user-attachments/assets/137f0f03-3212-4f81-bb40-2e19beb3c907)

If you think what I'm saying is a bit nit-picky, ignore this strange request.
if it's possible, please let me know 
I wanted to try it myself, but the fzf-lua repo is well encapsulated and it was difficult for me to find a place to start🥺

**Reply by ibhagwan** [reply]
I’m thinking where should such thing take place, maybe you can create an autocmd for `TermOpen` as the first event and then restore the cursor on either and autocmd for `FileType` with a condition for `fzf` (which fzf sets). Also try `winopts.on_create` lua call back (in fzf-lua’s setup), explore the sequence of all of these with prints which event comes before another and maybe some of these will help you achieve what you want.

**Comment by ibhagwan** [comment]
@Parsifa1, I edited the original answer, might be related to your issue:

> It also might be related to windows defender, some users are reporting tremendous boost after excluding the folder/executables (neovim, fzf, rg, etc)

**Comment by sibouras** [comment]
i wonder if a daemon helps, https://github.com/mantoni/core_d.js?tab=readme-ov-file#how-does-this-work

**Reply by ibhagwan** [reply]
> i wonder if a daemon helps, https://github.com/mantoni/core_d.js?tab=readme-ov-file#how-does-this-work

Avoiding the spawning of fzf will ceratinly help but not with everything, for example, triggering previews, reload actions, etc, all have to be triggered through a neovim `--headless` process running the lua code, parsing entries, removing icons, etc, without it you'd lose much of the functionality and remain "native" (no icons, no blings).

As for your suggesions, don't even need to go this far, fzf has `--listen`:
```man
       --listen[=[ADDR:]PORT] --listen-unsafe[=[ADDR:]PORT]
              Start HTTP server and listen on the given address. It allows
              external processes to send actions to perform via POST method.
```

It's just a lot of work (will need some refactoring) and debugging for Windows which I don't even use, last time I checked my donated Windows VPS was also dead so I'd have to find motivation for this.

**Reply by sibouras** [reply]
your explanation is helpful thanks, and ye and windows users are probably a minority so its not worth the effort

**Comment by phanen** [comment]
Related: https://github.com/neovim/neovim/pull/32846.
